cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,401 @@
+	tableActorRouter *actor.Router
+
+	pullerNode *pullerNode
+	sortNode   *sorterNode
+	sinkNode   *sinkNode
+
+	nodes   []*ActorNode
+	actorID actor.ID
+}
+
+// NewTableActor creates a table actor.
+func NewTableActor(cdcCtx cdcContext.Context,
+	mounter entry.Mounter,
+	tableID model.TableID,
+	tableName string,
+	replicaInfo *model.TableReplicaInfo,
+	sink sink.Sink,
+	targetTs model.Ts,
+) (TablePipeline, error) {
+	config := cdcCtx.ChangefeedVars().Info.Config
+	cyclicEnabled := config.Cyclic != nil && config.Cyclic.IsEnabled()
+	info := cdcCtx.ChangefeedVars()
+	vars := cdcCtx.GlobalVars()
+
+	actorID := vars.TableActorSystem.ActorID(info.ID, tableID)
+	mb := actor.NewMailbox(actorID, defaultOutputChannelSize)
+	// Cancel should be able to release all sub-goroutines in this actor.
+	ctx, cancel := context.WithCancel(cdcCtx)
+	// All sub-goroutines should be spawn in this wait group.
+	wg, cctx := errgroup.WithContext(ctx)
+	table := &tableActor{
+		// all errors in table actor will be reported to processor
+		reportErr: cdcCtx.Throw,
+		mb:        mb,
+		wg:        wg,
+		cancel:    cancel,
+
+		tableID:       tableID,
+		markTableID:   replicaInfo.MarkTableID,
+		tableName:     tableName,
+		cyclicEnabled: cyclicEnabled,
+		memoryQuota:   serverConfig.GetGlobalServerConfig().PerTableMemoryQuota,
+		mounter:       mounter,
+		replicaInfo:   replicaInfo,
+		replConfig:    config,
+		sink:          sink,
+		targetTs:      targetTs,
+		started:       false,
+
+		changefeedVars:   info,
+		globalVars:       vars,
+		tableActorRouter: vars.TableActorSystem.Router(),
+		actorID:          actorID,
+	}
+
+	startTime := time.Now()
+	log.Info("spawn and start table actor",
+		zap.String("changfeed", info.ID),
+		zap.String("tableName", tableName),
+		zap.Int64("tableID", tableID))
+	if err := table.start(cctx); err != nil {
+		table.stop(ctx, err)
+		return nil, errors.Trace(err)
+	}
+	err := vars.TableActorSystem.System().Spawn(mb, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	log.Info("spawn and start table actor done",
+		zap.String("changfeed", info.ID),
+		zap.String("tableName", tableName),
+		zap.Int64("tableID", tableID),
+		zap.Duration("duration", time.Since(startTime)))
+	return table, nil
+}
+
+func (t *tableActor) Poll(ctx context.Context, msgs []message.Message) bool {
+	for i := range msgs {
+		if t.stopped {
+			// No need to handle remaining messages.
+			break
+		}
+
+		switch msgs[i].Tp {
+		case message.TypeBarrier:
+			t.sortNode.UpdateBarrierTs(msgs[i].BarrierTs)
+			if err := t.sinkNode.UpdateBarrierTs(ctx, msgs[i].BarrierTs); err != nil {
+				t.stop(ctx, err)
+			}
+		case message.TypeTick:
+			_, err := t.sinkNode.HandleMessage(ctx, pipeline.TickMessage())
+			if err != nil {
+				t.stop(ctx, err)
+			}
+		case message.TypeStopSink:
+			// async stop the sink
+			go func() {
+				_, err := t.sinkNode.HandleMessage(ctx,
+					pipeline.CommandMessage(&pipeline.Command{Tp: pipeline.CommandTypeStop}),
```

Comment:
Keep line width within 80 chars, also for other lines in this file, thanks!

Reply:
fixed